api/v1alpha1/baseboardmanagement_types.go

```diff
@@ -17,29 +17,87 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// PowerState represents power state the BaseboardManagement.
+type PowerState string
+
+// BootDevice represents boot device of the BaseboardManagement.
+type BootDevice string
+
+// BaseboardManagementConditionType represents the condition of the BaseboardManagement.
+type BaseboardManagementConditionType string
+
+const (
+	On  PowerState = "on"
+	Off PowerState = "off"
+)
+
+const (
+	PXE   BootDevice = "pxe"
+	Disk  BootDevice = "disk"
+	BIOS  BootDevice = "bios"
+	CDROM BootDevice = "cdrom"
+	Safe  BootDevice = "safe"
+)
+
+const (
+	// ConnectionError represents failure to connect to the BaseboardManagement.
+	ConnectionError BaseboardManagementConditionType = "ConnectionError"
+)
+
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // BaseboardManagementSpec defines the desired state of BaseboardManagement
 type BaseboardManagementSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of BaseboardManagement. Edit baseboardmanagement_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Connection represents the BaseboardManagement connectivity information.
+	Connection Connection `json:"connection"`
+
+	// Power is the desired power state of the BaseboardManagement.
+	// +kubebuilder:validation:Enum=On;Off
+	Power PowerState `json:"power"`
+}
+
+type Connection struct {
```

Comment:
Needs a code comment. Is `Connection` the right name?

Reply:
I was debating with the name as well.
I thought of keeping it simple and going with `connection` because the struct has information targeting that.